sdk/python/lib/pulumi/provider/experimental/ComponentProvider.md

```diff
@@ -0,0 +1,185 @@
+# Component Provider
+
+The `pulumi.provider.experimental` package allows writing a Pulumi provider in Python with minimal boilerplate, using type annotations to define the schema. The provider can be published to Git or referenced locally.
+
+## Building a Component Provider
+
+Create a new directory that will hold the provider implementation.
+
+First we need to let Pulumi's plugin system know which language runtime should be used to run the provider plugin. Create a a `PulumiPlugin.yaml` file with the contents:
+
+```yaml
+runtime: python
+```
+
+> [!NOTE]
+> Python component providers only support the pip toolchain with a virtual environment named `venv`. Don't add any other runtime options to the `PulumiPlugin.yaml` file.
```

Comment:
Where have we landed on that? Will we support other tools or do we want to impose pip?